appservice/src/errors.ts

```diff
@@ -23,3 +25,31 @@ export class ArgumentError extends Error {
         super(localize('azFunc.argumentError', 'Invalid {0}.', obj.constructor.name));
     }
 }
+
+// tslint:disable-next-line:max-classes-per-file
+export class GitNotInstalledError extends Error {
```

Comment:
I'm not sure we want to expose this error outside the package either. Maybe just show the install prompt and return undefined from the function